src/mod_bosh.erl

```diff
@@ -160,12 +161,13 @@ node_cleanup(Acc, Node) ->
 
 -type option() :: {atom(), any()}.
 -spec init(req(), _Opts :: [option()]) -> {cowboy_loop, req(), rstate()}.
-init(Req, _Opts) ->
+init(Req, Opts) ->
     ?LOG_DEBUG(#{what => bosh_init, req => Req}),
     Msg = init_msg(Req),
     self() ! Msg,
+    CustomHeaders = gen_mod:get_opt(custom_headers, Opts, []),
     %% Upgrade to cowboy_loop behaviour to enable long polling
-    {cowboy_loop, Req, #rstate{}}.
+    {cowboy_loop, Req, #rstate{custom_headers = CustomHeaders}}.
```

Comment:
I think we can use `cowboy_req:set_resp_headers` here instead of passing `State` everywhere.